src/main/java/htsjdk/samtools/reference/FastaReferenceWriterBuilder.java

```diff
@@ -127,9 +155,18 @@ public FastaReferenceWriterBuilder setBasesPerLine(final int basesPerLine) {
     /**
      * Set the output index file to write to. Doesn't (currently) support compressed filenames.
      */
-    public FastaReferenceWriterBuilder setIndexFile(final Path indexFile) {
-        this.indexFile = indexFile;
-        this.indexOutput = null;
+    public FastaReferenceWriterBuilder setIndexFile(final Path faiIndexFile) {
+        this.faiIndexFile = faiIndexFile;
+        this.faiIndexOutput = null;
+        return this;
```

Comment:
I vetted all of the javadocs in this class for out of date allusions to compressed output. It is entirely acceptable behavior for a user to have a fasta.gz file with both a gzi index and a fai index. There is code to handle both of those indexes together invisibly elsewhere. Note I didn't update `setIndexFile()` -> `setFaiIndexFile()` because that seemed like it would be likely to break compatibility to dependent projects when most of the rest of this branch was purely additive.